src/org/zaproxy/zap/extension/openapi/generators/HeadersGenerator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.openapi.generators;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpHeaderField;
+import org.zaproxy.zap.extension.openapi.converter.swagger.OperationModel;
+
+import io.swagger.models.Operation;
+import io.swagger.models.parameters.HeaderParameter;
+import io.swagger.models.parameters.Parameter;
+
+public class HeadersGenerator {
+
+    private static final String ACCEPT = "Accept";
+    private static final String HEADER = "header";
+
+    public static List<HttpHeaderField> generate(OperationModel operationModel) {
+        List<HttpHeaderField> headerList = new LinkedList<HttpHeaderField>();
+        generateAcceptHeaders(operationModel.getOperation(), headerList);
+        generateContentTypeHeaders(operationModel.getOperation(), headerList);
+        generateCustomHeader(operationModel.getOperation(), headerList);
+        return headerList;
+    }
+
+    private static void generateCustomHeader(Operation operation, List<HttpHeaderField> headers) {
+        if (operation.getParameters() != null) {
+            for (Parameter parameter : operation.getParameters()) {
+                if (HEADER.equals(parameter.getIn())) {
+                    String name = parameter.getName();
+                    String value = DataGenerator.generate(((HeaderParameter) parameter));
+                    HttpHeaderField header = new HttpHeaderField(name, value);
+                    headers.add(header);
+                }
+            }
+        }
+    }
+
+    private static void generateContentTypeHeaders(Operation operation, List<HttpHeaderField> headers) {
+        if (operation.getConsumes() != null) {
+            for (String type : operation.getConsumes()) {

```

Comment:
It should check and use the Content-Type that ZAP produces, otherwise it might lead to the wrong Content-Type being set to the request, for example:
```JSON
"consumes":["application/json","application/xml"]
```
leads to the use of `application/xml`, which does not match the actual content sent (JSON).

Worth noting that in some requests there was no Content-Type header at all :/